libs/langgraph/langgraph/graph/schema_utils.py

```diff
@@ -62,18 +62,26 @@ def coerce(self, input_data: Any, depth: Optional[int] = None) -> Any:
         processed = {}
         if self._field_coercers is None:
             self._field_coercers = {
-                n: self._build_coercer(t) for n, t in self._fields.items()
+                n: self._build_coercer(t, depth - 1) for n, t in self._fields.items()
             }
         for k, v in input_data.items():
             fn = self._field_coercers.get(k)
             processed[k] = fn(v, depth - 1) if fn else v
         return self._construct(**processed)
 
-    def _build_coercer(self, field_type: Any) -> Callable[[Any, Any], Any]:
+    def _build_coercer(
+        self, field_type: Any, depth: int, throw: bool = False
+    ) -> Callable[[Any, Any], Any]:
+        if depth == 0:
+            return self._passthrough
+        if isinstance(field_type, ForwardRef):
+            resolved_type = field_type._evaluate(None, None, recursive_guard=frozenset())
```

Comment:
@nfcampos this is to get the RecurseModel definition coerced properly in `test_nested_pydantic_models`